### ISSUE TITLE
Conveying Water Use Data

### ISSUE THREAD

**Issue by amandaaprahamian** [California-Data-Collaborative/CA-Water-Data-Challenge#5]
Great job so far on the [Urban Drool Tool](https://codelaboc.github.io/CA-Stormwater-Data-Challenge/)!

Attached is a rough mock-up of what the layout of the Urban Drool Tool could look like. However the layout turns out, the basic data to convey is 
(1) spatially: where the census block boundaries, outfalls, and outfall drainage areas are and 
(2) with text: what is going on (water use, competitions, savings, and outfall info) in the census block the user is in.

![urbandrooltool](https://cloud.githubusercontent.com/assets/18447874/23858015/6dc7c486-07bc-11e7-8a66-9b9ef701644e.png)

**Comment by amandaaprahamian** [comment]
How do we want to use the water use data?
It might be difficult for someone to find meaning in a large water use number if their census block is large. Maybe we could do a per-capita water use number?
This number would be added to the side bar that populates with text on the Urban Drool Tool.

**Comment by patwater** [comment]
Per capita is a great idea!

**Comment by leighphan** [comment]
@amandaaprahamian 👍 to a per-capita water use number. Would anyone at CADC or MNWD be able to provide the data so it's readily available for input for CodeLab?

**Comment by monobina** [comment]
@leighphan @amandaaprahamian we were thinking it might be good to associate inefficient usage with more runoff. Inefficient usage would be Water Usage > Water Budget. I agree that it would make more sense to have per capita usage and kind of compare it with the per capita water budget maybe?

**Comment by monobina** [comment]
I think the water usage in the data is currently total for the census block. Not sure. 
@patwater will you be able to confirm what's the current unit of the usage data - total for census block or per capita?

**Comment by christophertull** [comment]
Pretty sure the usage is total use. But if the numbers you have are in the single or double digits then you probably have per capita use. 

Also the question of what to display is a good one. Will probably need to explore a bit to see if anything correlates with runoff.

* Total runoff correlated total usage would be expected, but because this is non-normalized might be less interesting. Finding blocks that deviate from this correlation (above or below expected) could be interesting though.

* per capita runoff vs. per capita usage might be more interesting. Would also hope for / expect some sort of correlation here.

* I think @monobina is right about trying to correlate inefficient usage with more runoff. Could be very cool to look at runoff as a a percent of water budget, or maybe better would be runoff as percent of total usage! This would show areas where a large chunk of their water use is being wasted.
  - Follow up would be to see if areas with with a greater percentage of runoff are more inefficient. E.g. plot (runoff as percent of usage) against (percent over/under budget)

**Comment by datwater** [comment]
Total water use might be misleading as runoff is caused with over-irrigation.  For example, a household might have a large lot and irrigate a lot more than someone with a small lot but the small lot might be applying more than what is needed (hence small lot is causing runoff).  Per capita will also bias towards less dense areas.  The key for identifying waste/ dry year runoff is over irrigation which requires both usage and comparing that to the water budget.  See here for an explanation of how it's calculated: https://www.mnwd.com/understandingwaterbudget/

The two metrics I think that make sense are:
1) total usage over budget summed up across all customers in a census block
2) weighted average percentage of budget ie:
(usage_1^2/waterbudget_1 + usage_2^2/waterbudget_2 ) / (usage_1 + usage_2)

The first shows the quantity that is potentially running off.  The 2nd shows the average efficiency.  For the 2nd, it would probably also help to understand the variance/ distribution and add that if possible.

**Comment by monobina** [comment]
I agree with @datwater on the metrics for measuring inefficiency. Per capita might be biased and total usage might be due to indoor use also. MNWD can help with setting up this metrics in the data or maybe CADC @patwater @christophertull can build these metrics with their cleaned up data ?

**Comment by patwater** [comment]
We're highly booked at the moment so would be best if came from MNWD :) 

@datwater what's the logic with raising usage to the second power in your weighted average out of curiousity?

**Comment by monobina** [comment]
Sure we can help with the data. Might have some questions for you in the process .

**Comment by datwater** [comment]
Its the formula to weight by usage:
usage_1/waterbudget_1 is the efficiency of customer 1.  If I want to take the weighted average efficiency I multiple each respective customer by their usage and divide by the total usage.

**Comment by datwater** [comment]
What we need from CaDC is the cleaned data linked to  census block groups.  Is there a way you can give Monobina access to MNWD data that you have cleaned?  She can develop the formula to apply key thing we need is the census block for each account.

**Comment by patwater** [comment]
@christophertull thoughts?  Correct me if I'm wrong though that should be a simple export.

**Comment by christophertull** [comment]
Yeah should be simple. Do you want customer data with census block? Or data
pre-aggregated to census blocks?

**Comment by monobina** [comment]
I think customer data with census blocks should work fine. I can then aggregate it how Drew suggested above.

**Comment by monobina** [comment]
@leighphan @amandaaprahamian we have the weighted average efficiency and total usage over budget by census block in the data now. Where do I share the data with you? Should I upload it in this repo?

**Comment by leighphan** [comment]
@monobina @amandaaprahamian Sure! Just curious, what is the time-frame for this data?

Could you upload the data to this original repo? Thanks!

**Comment by monobina** [comment]
@leighphan the data starts from Jan,2011 and is upto Aug,2016. Looks like i do not have push access to this repo. @christophertull @patwater can you please set me up with the push access to this repo so that I can upload the data? Thanks!

**Comment by christophertull** [comment]
@monobina done!

**Comment by monobina** [comment]
Thanks @christophertull !

**Comment by monobina** [comment]
@leighphan @amandaaprahamian I have uploaded the data in this repo in the Data folder. There are two csv files - TotalInefficiency and WeightedAverageEfficiency. TotalInefficiency provides the usage over budget (i.e. inefficient usage) by census block , year and month. The WeightedAverageEfficiency provides the average efficiency by census block. It is calculated using this formulae below which provides the weighted average of efficient usage in a census block
(usage_1^2/waterbudget_1 + usage_2^2/waterbudget_2 ) / (usage_1 + usage_2)

**Comment by monobina** [comment]
It will be great to have all your thoughts on Issue#6 "Using Inefficiency Data".

**Comment by patwater** [comment]
Added some thoughts in that Issue :)
